hw/ip/otbn/data/otbn.ld.tpl

```diff
@@ -52,7 +52,16 @@ SECTIONS
 
         *(.data*)
         . = ALIGN(32);
+
+        /* Align section end (see note in .text section) */
+        . = ALIGN(4);
+
+    } >dmem AT>dmem_load
+
+    .bss : ALIGN(32)
```

Comment:
I think we should add a comment here explaining that this section is for input/output buffers and why `_dmem_end` is defined here instead of the `.data` section above.

I'm not very familiar with otbn build process but it's not immediately clear to me why `.bss` does not end up in the final binary since it is assigned to the same memory region as `.data` (`>dmem AT>dmem_load`). `otbn_build.py` explicitly removes `.scratchpad` but I couldn't find any references to `.bss`. Also, should this section be `NOLOAD`? @rswarbrick

Reply:
I would have expected `.bss` to end up in the final binary, but it will still be a BSS section (as seen by the linker for the host binary): maybe we're managing to end up with memsz > filesz in the resulting host binary? This will be heavily dependent on where the OTBN image lands in the Ibex image.

To get a handle on what's going on, try linking the Ibex binary with `-Map`, which should show which input section is ending up where.

In particular, there's no magic handling for ".text" or ".data" in otbn_build.py. The only special section is `.scratchpad`, which gets chopped out.

Reply:
`_dmem_bss_end` is clear enough, resolving this.